### ISSUE TITLE
PHP: xmlreader extension is limited due dom extension not available at build time

### ISSUE THREAD

**Issue by edibopp** [NixOS/nixpkgs#93647]
**Describe the bug**
The xmlreader extension of PHP has a limited API surface due to missing DOM support.

**To Reproduce**

1. Create a `shell.nix` like this:

```nix
{ pkgs ? import <nixos-unstable> {} }:

let
  php = (pkgs.php.withExtensions ( { all, enabled }:
    enabled ++ (with all; [
      dom
      xmlreader
    ])
  ));

in pkgs.mkShell {
  buildInputs = [
    php
  ];
}

```

2. Use the dom-related functionality of xmlreader in a script `bug.php` like this:

```php
<?php
$reader = new XMLReader();
$reader->expand();
```

3. Run using `php bug.php`. You should see this warning:

```
Warning: DOM support is not enabled in <REDACTED>/bug.php on line 3
```

**Expected behavior**
DOM support is enabled and the xmlreader API works as intended.

**Additional context**
There is a very similar issue here: https://github.com/joyent/pkgsrc-legacy/issues/477

The warning looks like it originates from here:
https://github.com/php/php-src/blob/PHP-7.4.7/ext/xmlreader/php_xmlreader.c#L1154

This indicates that the `HAVE_DOM` compiler flag is not set when building the xmlreader extension. I don't know enough about the PHP packages in nixpkgs to fix this myself, but I hope this helps in narrowing down the issue.

**Notify maintainers**
@aanderse @etu @globin @Ma27 @talyz 

**Metadata**

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.50, NixOS, 20.03.2425.9a3e34ae601 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20200622_334e26b`
 - channels(root): `"nixos-20.03.2425.9a3e34ae601, nixos-unstable-20.09pre232864.55668eb671b"`
 - channels(eduard): `"home-manager-20.03, release-18.09.tar.gz-18.09"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by aanderse** [comment]
@talyz do you have any capacity to take a look at this?

**Comment by aanderse** [comment]
Will have to confirm if this is still relevant and then take a look.

**Comment by edibopp** [comment]
At least I can still reproduce it on latest nixpkgs.

**Comment by talyz** [comment]
Sorry that I missed this. Thanks for the detailed report! I think I've fixed this in #111317; I would appreciate if you could try it out.
